app/src/main/java/com/sun_asterisk/moviedb_50/data/source/remote/Repository.kt

```diff
@@ -0,0 +1,29 @@
+package com.sun_asterisk.moviedb_50.data.source.remote
+
+import com.sun_asterisk.moviedb_50.data.model.Genres
+import com.sun_asterisk.moviedb_50.data.model.Movie
+import com.sun_asterisk.moviedb_50.data.source.DataSource
+
+class Repository private constructor(
+    private val remoteDataSource: DataSource.RemoteDataSource,
+    private val localDataSource: DataSource.LocalDataSource
+) {
+    fun getGenres(onFetchDataJsonListener: OnFetchDataJsonListener<MutableList<Genres>>?) {
+        remoteDataSource.getGenres(onFetchDataJsonListener)
+    }
+
+    fun getMovies(
+        onFetchDataJsonListener: OnFetchDataJsonListener<MutableList<Movie>>?,
+        type: String
```

Comment:
```suggestion
    fun getMovies(
        type: String,
        onFetchDataJsonListener: OnFetchDataJsonListener<MutableList<Movie>>?
```
param listener luôn để cuối nha